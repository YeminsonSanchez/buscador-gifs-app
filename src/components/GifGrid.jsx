import { GifItem } from "./GifItem";
import { useFetchGifts } from "../hooks/useFetchGifts";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifts(category);

  return (
    <>
      <h3>{category}</h3>
      {/* si isLoading esta en tru ejecuta la segunda parte del && de lo contrario no hace nada */}
      {isLoading && <h4>Cargando...</h4>}
      <div className="card-grid">
        {images.map((image) => (
          // (...image) se usa para enviar todas las propiedades de la image al componente
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
