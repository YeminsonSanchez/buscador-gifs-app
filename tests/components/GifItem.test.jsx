import { getByRole, render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("pruebas en GifItem", () => {
  const title = "Saitama";
  const url = "https://media1.giphy.com/media/l0Hl6qZqQXqQ/200w.gif";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostar la imagen con el url indicado y el alt", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    // screen.debug();

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe mostar el titulo con el texto indicado", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getAllByText(title)).toBeTruthy();
  });
});
