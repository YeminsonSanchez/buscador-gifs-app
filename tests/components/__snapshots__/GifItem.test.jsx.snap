// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media1.giphy.com/media/l0Hl6qZqQXqQ/200w.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
